### ISSUE TITLE
support extending `angularCompilerOptions` in tsconfig just like cli

### ISSUE THREAD

**Issue by JounQin** [angular/vscode-ng-language-service#1126]
**Is your feature request related to a problem? Please describe.**

![image](https://user-images.githubusercontent.com/8336744/108317654-92493a00-71f9-11eb-8524-0a8d9931a744.png)

**Describe the solution you'd like**

I have a base tsconfig with correct `angularCompilerOptions`, but this plugin is still reporting missing `strictTemplates` option.

**Describe alternatives you've considered**

As title, align angular cli.

**Additional context**

N/A

**Comment by Muromi-Rikka** [comment]
are you use nx-workspace?

**Comment by JounQin** [comment]
> are you use nx-workspace?

Sorry, but what is nx-workspace?

**Comment by JounQin** [comment]
related https://github.com/angular/angular/issues/36715

**Comment by kyliau** [comment]
This has been fixed in Angular upstream, but only in version 12, so will only be available around May 2021.

**Comment by JounQin** [comment]
@kyliau Another question here, does this plugin read `angular.json` to get the correct `tsconfig.json` used or it just read `<project dir>/tsconfig.json`?

**Comment by kyliau** [comment]
@JounQin The Angular extension reads tsconfig.json from project dir. You can open the log file by running the vscode command `Open Angular server log` to see which tsconfig.json is loaded.

**Comment by JounQin** [comment]
> @JounQin The Angular extension reads tsconfig.json from project dir. You can open the log file by running the vscode command `Open Angular server log` to see which tsconfig.json is loaded.

What's your meaning by `project dir`? Is it read from `angular.json`? As I can tell, it doesn't read `angular.json` correctly to me.

**Comment by kyliau** [comment]
> What's your meaning by `project dir`? Is it read from `angular.json`? As I can tell, it doesn't read `angular.json` correctly to me.

The Angular extension will traverse up the directory tree to look for `tsconfig.json`. If it finds one, and if the `tsconfig.json` includes the source file, then it will use the `tsconfig.json` to create a project.
`angular.json` is purely an Angular CLI thing, the extension does not read it in any way nor is it aware of the file.

**Comment by kyliau** [comment]
Update: fix has been cherry picked to v11.2.x, will be released this week
https://github.com/angular/angular/pull/41036

**Comment by kyliau** [comment]
Fix released in v11.2.5

**Comment by JounQin** [comment]
@kyliau @alan-agius4 I tried the latest version `"@angular/core": "^11.2.7"`, but it's not working still

![image](https://user-images.githubusercontent.com/8336744/112424421-544ab300-8d6f-11eb-9ae4-36eb0675d5e7.png)

My tsconfig is:

```jsonc
{
  "extends": "@1stg/tsconfig/angular",
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "*": ["src/*"]
    }
  }
}
```
